src/taskgraph/generator.py

```diff
@@ -328,15 +328,19 @@ def _run(self):
         edges = set()
         for t in full_task_set:
             for depname, dep in t.dependencies.items():
-                edges.add((t.label, dep, depname))
+                if dep not in all_tasks.keys():
+                    raise Exception(
+                        f" Task '{t.label}' lists a dependency that does not exist: '{dep}'"
+                    )
+                else:
```

Comment:
nitpick: Please remove this `else` clause as there's no need for it.

Reply:
```
Task 'doc-generate' lists a dependency that does not exist: 'build-linux'
Traceback (most recent call last):
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/main.py", line 828, in main
    args.command(vars(args))
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/main.py", line 411, in show_taskgraph
    generate_taskgraph(options, parameters, logdir)
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/main.py", line 190, in generate_taskgraph
    out = format_taskgraph(options, spec, logfile(spec))
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/main.py", line 146, in format_taskgraph
    tg = getattr(tgg, options["graph_attr"])
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/generator.py", line 171, in full_task_graph
    return self._run_until("full_task_graph")
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/generator.py", line 424, in _run_until
    k, v = next(self._run)
  File "/Users/aakashsharma/Documents/projects/taskgraph/src/taskgraph/generator.py", line 337, in _run
    raise Exception("Tasks have some dependencies that does not exist: ")
Exception: Tasks have some dependencies that does not exist:

```
@ahal for -> Refactor this so it detects all invalid dependencies for all tasks and prints every single one of them at once (this way we avoid the scenario where someone fixes one broken dependency, only to hit the next one)
pls check above if above changes are fine or not

Reply:
Yeah the message looks good though, you'll need to actually list the dependencies afterwards. I'd think something like:
```
Exception: Some tasks have dependencies that do not exist!

  doc-generate depends on:
    - build-bar
    - build-foo

  build-linux depends on:
    - invalid-task
```